google/resource_compute_address.go

```diff
@@ -83,55 +99,107 @@ func resourceComputeAddressCreate(d *schema.ResourceData, meta interface{}) erro
 	config := meta.(*Config)
 
-	region, err := getRegion(d, config)
+	addressId, err := parseComputeAddressId(d.Id(), config)
 	if err != nil {
 		return err
 	}
 
-	project, err := getProject(d, config)
-	if err != nil {
-		return err
-	}
+	log.Println("[rosbo] resourceComputeAddressRead", addressId)
 
 	addr, err := config.clientCompute.Addresses.Get(
-		project, region, d.Id()).Do()
+		addressId.Project, addressId.Region, addressId.Name).Do()
 	if err != nil {
 		return handleNotFoundError(err, d, fmt.Sprintf("Address %q", d.Get("name").(string)))
 	}
 
 	d.Set("address", addr.Address)
 	d.Set("self_link", addr.SelfLink)
 	d.Set("name", addr.Name)
+	d.Set("region", addr.Region)
 
 	return nil
 }
 
 func resourceComputeAddressDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	region, err := getRegion(d, config)
-	if err != nil {
-		return err
-	}
-
-	project, err := getProject(d, config)
+	addressId, err := parseComputeAddressId(d.Id(), config)
 	if err != nil {
 		return err
 	}
 
 	// Delete the address
 	log.Printf("[DEBUG] address delete request")
 	op, err := config.clientCompute.Addresses.Delete(
-		project, region, d.Id()).Do()
+		addressId.Project, addressId.Region, addressId.Name).Do()
 	if err != nil {
 		return fmt.Errorf("Error deleting address: %s", err)
 	}
 
-	err = computeOperationWait(config, op, project, "Deleting Address")
+	err = computeOperationWait(config, op, addressId.Project, "Deleting Address")
 	if err != nil {
 		return err
 	}
 
 	d.SetId("")
 	return nil
 }
+
+func resourceComputeAddressImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
+
+	addressId, err := parseComputeAddressId(d.Id(), config)
+	if err != nil {
+		return nil, err
+	}
+
+	d.SetId(addressId.canonicalId())
+
+	return []*schema.ResourceData{d}, nil
+}
+
+type computeAddressId struct {
+	Project string
+	Region  string
+	Name    string
+}
+
+func (s computeAddressId) canonicalId() string {
+	return fmt.Sprintf(computeAddressIdTemplate, s.Project, s.Region, s.Name)
+}
+
+func parseComputeAddressId(id string, config *Config) (*computeAddressId, error) {
+	var parts []string
+	if computeAddressLinkRegex.MatchString(id) {
+		parts = computeAddressLinkRegex.FindStringSubmatch(id)
+
+		return &computeAddressId{
+			Project: parts[1],
+			Region:  parts[2],
+			Name:    parts[3],
+		}, nil
+	} else {
+		parts = strings.Split(id, "/")
+	}
+
+	if len(parts) == 3 {
+		return &computeAddressId{
+			Project: parts[0],
+			Region:  parts[1],
+			Name:    parts[2],
+		}, nil
+	} else if len(parts) == 2 {
+		// Project is optional.
+		if config.Project == "" {
+			return nil, fmt.Errorf("Invalid compute address id. The default project for the provider must be set.")
+		}
+
+		return &computeAddressId{
+			Project: config.Project,
+			Region:  parts[0],
+			Name:    parts[1],
+		}, nil
+	}
```

Comment:
Missing one more case (only specifying `name`, which is the current behavior), or was this not included on purpose?

Reply:
It was on purpose but now that I think about it, I see no good reason for not supporting this case. Support added.